commerce_coordinator/apps/iap/api/v1/views.py

```diff
@@ -0,0 +1,188 @@
+
+        data = serializer.validated_data
+
+        cart_id = data["cart_id"]
+        order_number = data["order_number"]
+        payment_method = data["payment_method"]
+        shipping_address_data = data.get("shipping_address")
+        paypal_order = data.get("paypal_order") or {
+            "id": "TEST12345",
+            "status": "APPROVED",
+            "paymentSource": {"type": payment_method or "paypal"}
+        }
+
+        try:
+            ct_client = CommercetoolsAPIClient()
+
+            cart = ct_client.base_client.carts.get_by_id(cart_id)
+
+            if not cart.customer_id or str(cart.customer_id) != str(request.user.customer_id):
+                logger.warning(
+                    f"[CreateOrderView] Unauthorized cart access attempt by user "
+                    f"[{request.user.id}] on cart [{cart.id}]."
+                )
+                return JsonResponse({'error': 'Unauthorized access to cart.'}, status=status.HTTP_403_FORBIDDEN)
+
+            if not is_cart_active(cart):
+                logger.error(f"[CreateOrderView] Cart {cart_id} is not active. Current state: {cart.cart_state}")
+                return Response(
+                    {"error": f"Cart {cart_id} is not in active state. Current state: {cart.cart_state}"},
+                    status=status.HTTP_400_BAD_REQUEST
+                )
+
+            try:
+                cart = set_shipping_address(cart, shipping_address_data, ct_client)
+            except ValueError as e:
+                logger.error(f"[CreateOrderView] ValueError encountered: {str(e)}")
+                return Response({"error": "Invalid shipping address"}, status=status.HTTP_400_BAD_REQUEST)
+            except CommercetoolsError as e:
+                logger.error(f"[CreateOrderView] Error setting shipping address: {str(e)}")
+                return Response({"error": "Failed to process shipping address."},
+                                status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+            payment_source = paypal_order.get("paymentSource")
+            method_key = payment_source.get("type") if payment_source else payment_method
+            amount_planned = cart.total_price
+            payment_method_info = PaymentMethodInfo(
+                payment_interface=PAYPAL_PAYMENT_SERVICE_PROVIDER,
+                method=method_key,
+                name={"en": method_key},
+            )
+            payment_status = PaymentStatusDraft(interface_code=paypal_order.get("status"))
+            payment_draft = PaymentDraft(
+                key=paypal_order.get("id"),
+                amount_planned=amount_planned,
+                payment_method_info=payment_method_info,
+                payment_status=payment_status,
+            )
+
+            try:
+                payment = ct_client.base_client.payments.create(payment_draft)
+            except CommercetoolsError as e:
+                logger.error(f"[CreateOrderView] Failed to create payment: {str(e)}")
```

Comment:
```suggestion
                logger.exception("[CreateOrderView] Failed to create payment: ")
```
logger.exception automatically log exception